### DISCUSSION TITLE
How to write a custom check for the same configuration block defined multiple times

### DISCUSSION THREAD

**Discussion by aleksandary** [aquasecurity/tfsec#1832] [Help and suppports]
Hello guys,

I am trying to write a custom check when we have the same configuration block defined multiple times (see below) and it seems that I need a little help with this one. 

So the Terraform code looks like this:

```hcl
resource "aws_elb" "bar" {

  listener {
    instance_port     = 8000
    instance_protocol = "http"
    lb_port           = 80
    lb_protocol       = "http"
  }

  listener {
    instance_port      = 8000
    instance_protocol  = "http"
    lb_port            = 443
    lb_protocol        = "https"
    ssl_certificate_id = "arn:aws:iam::123456789012:server-certificate/certName"
  }
```

I am trying to create the custom check to check if we have a listener block with a lb_protocol value `http` and if yes to check if there is a second listener block with lb_protocol value `https`.

I have searched in the documentation, the discussions here but didn't find anything similar. 

Would it be possible to configure a such check?

Thank you in advance.

Best regards,
Aleksandar Yosifov

**Comment by owenrumney** [comment]
I don't think anything like this exists in the available matchspecs. I can have a look at what would make sense to add.

Another option is to write the check in Rego which might give quicker results
